modules/setting/security.go

```diff
@@ -157,4 +158,7 @@ func loadSecurityFrom(rootCfg ConfigProvider) {
 			PasswordComplexity = append(PasswordComplexity, name)
 		}
 	}
+
+	// TODO: default value should be true immediately after 1.22.0 has been released, so that 1.23.0 ships with the change
+	DisableQueryAuthToken = sec.Key("DISABLE_QUERY_AUTH_TOKEN").MustBool(false)
```

Comment:
As lunny pointed out we should log a warning here if the value is false. Something like:

```
if DisableQueryAuthToken is false:
 warn "Enabling Query API Auth tokens is not recommended. DISABLE_QUERY_AUTH_TOKEN will default to true in gitea 1.23 and will be removed in gitea 1.24."
```

Reply:
You can use `deprecatedSetting` directly.